internal/contour/metrics.go

```diff
@@ -31,29 +30,18 @@ func calculateIngressRouteMetric(statuses map[dag.Meta]dag.Status) metrics.Ingre
 	metricRoots := make(map[metrics.Meta]int)
 
 	for _, v := range statuses {
-		var namespace string
-		switch obj := v.Object.(type) {
-		case *ingressroutev1.IngressRoute:
-			namespace = obj.Namespace
-			if obj.Spec.VirtualHost != nil {
-				metricRoots[metrics.Meta{Namespace: namespace}]++
-			}
-		case *projcontour.HTTPLoadBalancer:
-			namespace = obj.Namespace
-			if obj.Spec.VirtualHost != nil {
-				metricRoots[metrics.Meta{Namespace: namespace}]++
-			}
-		}
-
 		switch v.Status {
 		case dag.StatusValid:
-			metricValid[metrics.Meta{VHost: v.Vhost, Namespace: namespace}]++
+			metricValid[metrics.Meta{VHost: v.Vhost, Namespace: v.Object.GetObjectMeta().GetNamespace()}]++
 		case dag.StatusInvalid:
-			metricInvalid[metrics.Meta{VHost: v.Vhost, Namespace: namespace}]++
+			metricInvalid[metrics.Meta{VHost: v.Vhost, Namespace: v.Object.GetObjectMeta().GetNamespace()}]++
 		case dag.StatusOrphaned:
-			metricOrphaned[metrics.Meta{Namespace: namespace}]++
+			metricOrphaned[metrics.Meta{Namespace: v.Object.GetObjectMeta().GetNamespace()}]++
+		}
+		if v.Object.(*ingressroutev1.IngressRoute).Spec.VirtualHost != nil {
```

Comment:
Need to check for typei= of `v.Object` here, if I use an `HTTPLoadBalancer` I get an exception since the type isn't an `IngressRoute`.

Reply:
Ahh, thank you. Good catch, I need to type assert here.